database/engine/journalfile.h

```diff
@@ -13,37 +13,117 @@ struct rrdengine_journalfile;
 
 #define WALFILE_PREFIX "journalfile-"
 #define WALFILE_EXTENSION ".njf"
+#define WALFILE_EXTENSION_V2 ".njfv2"
 
+#define is_descr_journal_v2(descr) ((descr)->extent_entry != NULL)
 
 /* only one event loop is supported for now */
 struct rrdengine_journalfile {
     uv_file file;
     uint64_t pos;
     void *data;
+    void *journal_data;                         // MMAPed file of journal v2
+    uint32_t journal_data_size;                 // Total file size mapped
     struct rrdengine_datafile *datafile;
 };
 
+
+// Journal v2 structures
+
+#define JOURVAL_V2_MAGIC           (0x01221019)
+#define JOURVAL_V2_REBUILD_MAGIC   (0x00221019)
+#define JOURVAL_V2_SKIP_MAGIC      (0x02221019)
+
+struct journal_v2_block_trailer {
+    union {
+        uint8_t checksum[CHECKSUM_SZ]; /* CRC32 */
+        uint32_t crc;
+    };
+};
+
+// Journal V2
+// 28 bytes
+struct journal_page_header {
+    union {
+        uint8_t checksum[4];      // CRC check
```

Comment:
I suggest to also use `CHECKSUM_SZ` instead `4` here, like we have in `journal_v2_block_trailer`.

Reply:
Fixed this too.